src/targets/maven.ts

```diff
@@ -0,0 +1,389 @@
+import { TargetConfig } from '../schemas/project_config';
+import {
+  BaseArtifactProvider,
+  RemoteArtifact,
+} from '../artifact_providers/base';
+import { BaseTarget } from './base';
+import { homedir } from 'os';
+import { basename, join, parse } from 'path';
+import { promises as fsPromises } from 'fs';
+import { sleep, withRetry } from '../utils/async';
+import {
+  checkExecutableIsPresent,
+  extractZipArchive,
+  spawnProcess,
+} from '../utils/system';
+import { withTempDir } from '../utils/files';
+import { checkEnvForPrerequisite } from '../utils/env';
+
+const GRADLE_PROPERTIES_FILENAME = 'gradle.properties';
+
+/**
+ * Maximum number of attempts including the initial one when publishing fails.
+ * After this number of retries, publishing fails.
+ */
+const MAX_PUBLISHING_ATTEMPTS = 5;
+
+/**
+ * Delay between retries of publish operations, in seconds.
+ */
+const RETRY_DELAY_SECS = 3;
+
+/**
+ * Exponential backoff applied to the retry delay.
+ */
+const RETRY_EXP_FACTOR = 2;
+
+const targetSecrets = [
+  'OSSRH_USERNAME',
+  'OSSRH_PASSWORD',
+  'MAVEN_CENTRAL_USERNAME',
+  'MAVEN_CENTRAL_PASSWORD',
+] as const;
+type SecretsType = typeof targetSecrets[number];
+
+const targetOptions = [
+  'androidDistDirPattern',
+  'androidFileReplaceePattern',
+  'androidFileReplacerStr',
+  'gradleCliPath',
+  'mavenCliPath',
+  'mavenSettingsPath',
+  'mavenRepoId',
+  'mavenRepoUrl',
+] as const;
+type OptionsType = typeof targetOptions[number];
+
+type TargetSettingType = SecretsType | OptionsType;
+
+/**
+ * Config options for the "maven" target.
+ */
+type MavenTargetConfig = {
+  [key in TargetSettingType]: string;
```

Comment:
`TargetSettingType` is used later (in `getTargetSecrets` and `getTargetSettings`), so I thought creating a type to avoid unions would help reading. Wdyt?

Reply:
Oh, right. Missed that sorry.